src/utils/FooterCalculator.js

```diff
@@ -0,0 +1,131 @@
+const DIRTYFIELD_TDPLAIN = 1
+const DIRTYFIELD_TDPCT = 2
+const TYPECHANGE = 1
+
+class DiscountCalculator {
+  constructor(
+    _tdAmount,
+    _tdPlain,
+    _tdPct,
+    _tdType,
+    _subtotal,
+    _currentDiscount,
+    _hiddenTdPct,
+    _hiddenTdAmount,
+    _typeChange
+  ) {
+    this.tdAmount = _tdAmount
+    this.tdPlain = _tdPlain
+    this.tdPct = _tdPct
+    this.tdType = _tdType
+    this.subtotal = _subtotal
+    this.currentDiscount = _currentDiscount
+    this.hiddenTdPct = _hiddenTdPct
+    this.hiddenTdAmount = _hiddenTdAmount
+    this.typeChange = _typeChange
+  }
+}
+
+class FooterSummary {
+  constructor(_totalQty, _totalWeight, _totalVolume, _totalUpo, _sumVat, _sumExtended, _tdAmount, _net, _miscAmount) {
+    this.totalQty = _totalQty
+    this.totalWeight = _totalWeight
+    this.totalVolume = _totalVolume
+    this.totalUpo = _totalUpo
+    this.sumVat = _sumVat
+    this.sumExtended = _sumExtended
+    this.tdAmount = _tdAmount
+    this.net = _net
+    this.miscAmount = _miscAmount
+  }
+}
+
+class FooterCalculator {
+  constructor(_array, _footerSummary) {
+    this.array = _array
+    this.footerSummary = _footerSummary
+  }
+}
+
+const returnFooterResults = _discountObj => {
+  if (_discountObj.tdPct === true) {
+    // If it's %
+    if (_discountObj.tdAmount > 100) {
```

Comment:
do we need comments ?

Reply:
I prefer to keep them, they will help us to understand the code quickly when we review this file later